src/platform/silabs/SilabsConfig.cpp

```diff
@@ -284,6 +284,11 @@ CHIP_ERROR SilabsConfig::WriteConfigValueStr(Key key, const char * str, size_t s
 
     if (str != NULL)
     {
+        // Compute the length of the string if not provided.
+        if (strLen == 0)
+        {
+            strLen = strlen(str);
```

Comment:
we should use `strnlen` instead to prevent buffer overflows right?

Reply:
Yes, but we aren't providing the size in this usecase... we could define a max string size but in the event of a string smaller than max, we could still overflow.

Reply:
Strnlen requires the length, since we don't know it here, that wouldn't be much help.